src/package.lisp

```diff
@@ -28,6 +28,19 @@
       package-designator
       (oget *package-table* (string package-designator))))
 
+(defun package-designator-package (package-designator)
+  "Return a package named by the package-designator."
+  (etypecase package-designator
+    (package package-designator)
+    ;; XXX: Merge cases when etypecase supports type specifiers, currently it
+    ;; only supports types
+    (string (find-package package-designator))
+    (symbol (find-package package-designator))))
+
+(defun delete-package (package-designator)
+  (delete-property (package-name (package-designator-package package-designator))
```

Comment:
Also, it would be cool if we could implement the `unuse-package` too and have a completely working of delete-package. If not, please add a TODO comment or an issue so we don't forget it.